bfd.cmake error - unset called with incorrect number of arguments

Hello,

First of all, it is great to see the recent surge in activity in this project :)

I was trying to compile the last version on Ubuntu 24.04, and I hit this error:

<pre>-- git revision rev. 2.3.3-207-gcdbc34c13 found
-- Looking for bfd.h
-- Looking for bfd.h - found
-- Performing Test BFD_COMPILE_TEST
-- Performing Test BFD_COMPILE_TEST - Success
CMake Error at cmake/bfd.cmake:29 (unset):
  unset called with incorrect number of arguments
Call Stack (most recent call first):
  CMakeLists.txt:137 (include)
</pre>



When checking bfd.cmake:29, we can see this:
`unset (${CMAKE_REQUIRED_LIBRARIES})`


I'm a newbie to cmake, so I tried changing the syntax to:
`unset (CMAKE_REQUIRED_LIBRARIES)`

And then I got:
<pre>-- git revision rev. 2.3.3-207-gcdbc34c13 found
-- Looking for bfd.h
-- Looking for bfd.h - found
-- Performing Test BFD_COMPILE_TEST
-- Performing Test BFD_COMPILE_TEST - Success
-- bfd.h found but can't link against it, disabling support
-- 

	Configured aMule SVN for 'Linux-6.17.0-22-generic' on 'x86_64'.

</pre>


The linking issue, seems to come from the empty "" element in the first position of the foreach loop. 

<pre>
foreach (CMAKE_REQUIRED_LIBRARIES
                               ""  "${LIBBFD_TMP}" "${LIBBFD_TMP};iberty" "${LIBBFD_TMP};dl" "${LIBBFD};iberty;dl" "${LIBBFD_TMP};${LIBINTL}" "${LIBBFD_TMP};iberty;${LIBINTL_TMP}" "${LIBBFD_TMP};iberty;dl;${LIBINBTL}"
</pre>

After removing it, I get:

<pre>-- git revision rev. 2.3.3-207-gcdbc34c13 found
-- Looking for bfd.h
-- Looking for bfd.h - found
-- Performing Test BFD_COMPILE_TEST
-- Performing Test BFD_COMPILE_TEST - Success
-- 

	Configured aMule SVN for 'Linux-6.17.0-22-generic' on 'x86_64'.

</pre>

And it builds correctly after that.

As i'm not an expert on cmake or the purpose of this library, i'm opening this issue to get your thoughts.
What shall be the best approach to fix this issue?

--- a/cmake/bfd.cmake
+++ b/cmake/bfd.cmake
@@ -11,7 +11,13 @@ if (NOT HAVE_BFD)
 			message (STATUS "No useable bfd-lib found, disabling support")
 		else()
 			foreach (CMAKE_REQUIRED_LIBRARIES
-				"" "${LIBBFD_TMP}" "${LIBBFD_TMP};iberty" "${LIBBFD_TMP};dl" "${LIBBFD};iberty;dl" "${LIBBFD_TMP};${LIBINTL}" "${LIBBFD_TMP};iberty;${LIBINTL_TMP}" "${LIBBFD_TMP};iberty;dl;${LIBINBTL}"
+				"${LIBBFD_TMP}"
+				"${LIBBFD_TMP};iberty"
+				"${LIBBFD_TMP};dl"
+				"${LIBBFD_TMP};iberty;dl"
+				"${LIBBFD_TMP};${LIBINTL}"
+				"${LIBBFD_TMP};iberty;${LIBINTL}"
+				"${LIBBFD_TMP};iberty;dl;${LIBINTL}"
 			)
 				unset (BFD_COMPILE_TEST CACHE)
 				check_c_source_compiles ("#include <ansidecl.h>
@@ -26,9 +32,9 @@ if (NOT HAVE_BFD)
 
 				if (BFD_COMPILE_TEST)
 					set (BFD_LIBRARY ${CMAKE_REQUIRED_LIBRARIES} CACHE STRING "Lib to use when linking to bfd")
-					unset (${CMAKE_REQUIRED_LIBRARIES})
+					unset (CMAKE_REQUIRED_LIBRARIES)
 					break()
-				endif()	
+				endif()
 			endforeach()
 		endif()
 

